### PR TITLE
Adding OpCert counter details

### PR DESCRIPTION
**Pull request by olgahryniuk**

### CODE REVIEW

**Comment by SViksha** [comment]
text says "Note that with the Vasil hard fork, you should create your operational certificate using cold.counter +1. This means that the counter must be exactly one more than the previously used one." , its not "exactly one more than previously use one" right ? can be more than +1?

File: doc/stake-pool-operations/KES_period.md [hunk 1]
```diff
@@ -44,3 +44,5 @@ With this we are able to generate an operational certificate for our stake pool:
     --operational-certificate-issue-counter cold.counter \
     --kes-period 205 \
     --out-file node.cert
+
+Note that with the Vasil hard fork, you should create your operational certificate using cold.counter +1. This means that the counter must be exactly one more than the previously used one.
```
Comment (JaredCorduan) [inline doc/stake-pool-operations/KES_period.md:48]:
```suggestion
Note that with the Vasil hard fork, you must create your operational certificate using cold.counter +1. This means that the counter must be exactly one more than the previously used one.
```

**Review by JaredCorduan — APPROVED** [review]
LGTM!

**Review by dcoutts — APPROVED** [review]

**Comment by btbf** [comment]
@dcoutts 
What led to the implementation of this mechanism?

Suppose the counter number of the last block generated is 1. 
Then it expires and the KES is renewed and the counter number is now 2. 
However, if there is no chance to generate a block and the KES is due for renewal, does this mean that the counter must be renewed as 2 instead of 3?

**Comment by CarlosLopezDeLara** [comment]
@olgahryniuk  Perhaps we should try to answer @dorin100 questions from-->  https://github.com/input-output-hk/cardano-node/issues/3959

**Comment by SViksha** [comment]
> > if there is no chance to generate a block and the KES is due for renewal, does this mean that the counter must be renewed as 2 instead of 3?
> 
> No, you'd renew with 2 as last block from that pool on chain would only have 1 as op cert Counter.
> 
> It is confusing for SPOs who may not be closely monitoring details and the reason for this change should be documented imo - especially as it complicates the workflow further for offline keys , and for online ones - CLI does not accept pool ID/VRF key as input to query counter info (#3959 #3976)

I would suggest adding this scenario in the release note, or changing the text to 

" This command will increment the number in the `cold.counter` file by one. Note that from the Vasil hard fork onwards this is required behaviour: it is only valid for new operational certificates to use the counter value (previously used on the chain ### with successful block production ) plus *exactly* one. Prior to Vasil, it was permitted to use any strictly larger counter value than used previously, but this is no longer permitted. It must be incremented by exactly one. In particular, this means one cannot use the current time or slot number as an issue number. "

**Comment by dcoutts** [comment]
Yeah, thanks folks, we do need to clarify this. Thanks for the questions and suggested wording.

Yes, the crucial point is that the op cert counter has actually appeared in a block on the chain. So it doesn't matter if one deployed an op cert but ended up not forging a block, or if a block got forged but lost a block height battle: those don't result in the op cert being in a block that is in the chain.

**Comment by olgahryniuk** [comment]
Thanks, everyone for your input! As agreed with @dcoutts we've tweaked the wording for better clarity. More details about commands will be added later and referenced in this document.
@Jimbo4350 will very much appreciate your final review/merging

**Comment by SViksha** [comment]
> Yeah, thanks folks, we do need to clarify this. Thanks for the questions and suggested wording.
> 
> Yes, the crucial point is that the op cert counter has actually appeared in a block on the chain. So it doesn't matter if one deployed an op cert but ended up not forging a block, or if a block got forged but lost a block height battle: those don't result in the op cert being in a block that is in the chain.

One more thing is unclear: when we should increment the counter, after submitting the updated proposal or the next epoch when proposal became active?

**Comment by olgahryniuk** [comment]
Closing this PR in favor of an expanded work being done in #4116

**Pull request closed without merging**